src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -92,30 +88,27 @@ public void deliveryComplete(IMqttDeliveryToken token) {
      * @throws MQTTClientException if unable to create client for the mqtt broker
      */
     @Inject
-    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService)
-            throws MQTTClientException {
+    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+                      ExecutorService executorService) throws MQTTClientException {
         this(topics, mqttClientKeyStore, executorService, null);
         // TODO: Handle the case when serverUri is modified
         try {
-            this.mqttClientInternal = new MqttClient(serverUri, clientId, dataStore);
+            this.mqttClientInternal = new MqttClient(brokerUri.toString(), clientId, dataStore);
         } catch (MqttException e) {
             throw new MQTTClientException("Unable to create an MQTT client", e);
         }
     }
 
-    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore, ExecutorService executorService,
-                         IMqttClient mqttClient) {
-        // serverUri should take precedence since brokerServerUri is deprecated
-        String tmpUri = Coerce.toString(
-                topics.findOrDefault(DEFAULT_BROKER_URI, KernelConfigResolver.CONFIGURATION_CONFIG_KEY,
-                        DEPRECATED_BROKER_URI_KEY));
-        this.serverUri = Coerce.toString(
-                topics.findOrDefault(tmpUri, KernelConfigResolver.CONFIGURATION_CONFIG_KEY,
-                        BROKER_URI_KEY));
+    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+                         ExecutorService executorService, IMqttClient mqttClient) throws MQTTClientException {
+        try {
+            this.brokerUri = BridgeConfig.getBrokerUri(topics);
+        } catch (URISyntaxException e) {
+            throw new MQTTClientException("Invalid brokerUri configuration", e);
```

Comment:
be careful about throwing, you need to ensure that whoever catches the exception will handle it nicely. For example, if the catcher uses `serviceErrored` that will cause the service to go into the errored state and retry. Since it is retrying with the same data we know it will fail which means that it will go into the broken state. Do we handle the broken state correctly? Meaning that if the config changes to a proper URI, do we exit from broken? That isn't guaranteed to work correctly depending on how the rest of this plugin is written.

Reply:
Will take a closer look, I saw exceptions being thrown in the other constructor so I let my guard down 😅